ArduCopter/GCS_Mavlink.cpp

```diff
@@ -1009,6 +1010,14 @@ MAV_RESULT GCS_MAVLINK_Copter::handle_command_long_packet(const mavlink_command_
         GCS_MAVLINK_Copter::convert_COMMAND_LONG_to_COMMAND_INT(packet, packet_int);
         return handle_command_pause_continue(packet_int);
     }
+
+    // waypoint navigation
+    case MAV_CMD_NAV_WAYPOINT: {
+        mavlink_command_int_t packet_int;
+        GCS_MAVLINK_Copter::convert_COMMAND_LONG_to_COMMAND_INT(packet, packet_int);
+        return handle_command_int_do_reposition(packet_int);
```

Comment:
Isn't the function you're calling here going to ascribe special meaning to param2 which `NAV_WAYPOINT` doesn't have, or has different meanings for?